addons/web_editor/static/src/js/wysiwyg/wysiwyg.js

```diff
@@ -2907,6 +2911,9 @@ const Wysiwyg = Widget.extend({
         this._currentClientId = this._generateClientId();
         this.odooEditor.collaborationSetClientId(this._currentClientId);
         this.resetValue(value);
+        if (this.odooEditor.document.getSelection().isCollapsed) {
+            this.odooEditor._updateToolbar(false);
+        }
```

Comment:
i think this code targets the discard scenario right? well the toolbar should be updated in [resetContent](https://github.com/odoo/odoo/blob/16.0/addons/web_editor/static/src/js/editor/odoo-editor/src/OdooEditor.js#L718) of OdooEditor instead of doing it in`resetEditor`